acipher/ta/acipher_ta.c

```diff
@@ -81,10 +110,21 @@ static TEE_Result cmd_enc(struct acipher *state, uint32_t pt,
 	outbuf = params[1].memref.buffer;
 	outbuf_len = params[1].memref.size;
 
-	res = TEE_AllocateOperation(&op, alg, TEE_MODE_ENCRYPT,
+	res = select_algo(params[3].value.a, &alg_num);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	if (params[2].value.a)
+		encrypt = TEE_MODE_ENCRYPT;
+	else
+		encrypt = TEE_MODE_DECRYPT;
+
+	res = TEE_AllocateOperation(&op, alg_num, encrypt,
 				    key_info.keySize);
 	if (res) {
-		EMSG("TEE_AllocateOperation(TEE_MODE_ENCRYPT, %#" PRIx32 ", %" PRId32 "): %#" PRIx32, alg, key_info.keySize, res);
+		EMSG("TEE_AllocateOperation(TEE_MODE_ENCRYPT, %#"
+		     PRIx32 ", %" PRId32 "): %#" PRIx32,
```

Comment:
The show argument do not match the order used in `TEE_AllocateOperation()`. Not a big issue but one may found that confusing.